301-subnet-driven-deployment/azuredeploy.json

```diff
@@ -0,0 +1,290 @@
+      },
+      "properties": {
+        "mode": "incremental",
+        "templateLink": {
+          "uri": "[concat(parameters('assetLocation'),'CreateADC.json')]",
+          "contentVersion": "1.0.0.1"
+        },
+        "parameters": {
+          "indx": { "value": "[copyIndex(1)]" },
+          "lbName": { "value": "[parameters('lbName')]" },
+          "publicIPAddressName": { "value": "[parameters('publicIPAddressName')]" },
+          "publicStartRdpPort": { "value": 5000 },
+          "vnetName": { "value": "[parameters('vnetName')]" },
+          "subnets": { "value": "[reference(parameters('vnetName')).outputs.vnetSubnets.value]" },
+          "storageAccountName": { "value": "[reference('LabStorage').outputs.storageAccountName.value]" },
+          "availabilitySetName": { "value": "AvSetDC" },
+          "vmNamePrefix": { "value": "dc" },
+          "nicNamePrefix": { "value": "nic" },
+          "domainName": { "value": "[parameters('domainName')]" },
+          "adminUserName": { "value": "[parameters('adminUserName')]" },
+          "adminPassword": { "value": "[parameters('adminPassword')]" },
+          "assetLocation": { "value": "[parameters('assetLocation')]" }
+        }
+      }
+    },
+    {
+      "apiVersion": "2015-01-01",
+      "name": "[concat('Members', copyIndex())]",
+      "type": "Microsoft.Resources/deployments",
+      "dependsOn": [
+        "Microsoft.Resources/deployments/RootFirstDC",
+        "Microsoft.Resources/deployments/setDNS"
+      ],
+      "copy": {
+        "name": "members",
+        "count": "[mul(length(variables('subnets')),2)]"
+      },
+      "properties": {
+        "mode": "Incremental",
+        "templateLink": {
+          "uri": "[concat(parameters('assetLocation'),'WinServ.json')]",
+          "contentVersion": "1.0.0.1"
+        },
+        "parameters": {
+          "indx": { "value": "[copyIndex()]" },
+          "lbName": { "value": "[parameters('lbName')]" },
+          "computerName": { "value": "srvMbmr" },
+          "publicStartRdpPort": { "value": 6000 },
+          "publicIPAddressName": { "value": "[parameters('publicIPAddressName')]" },
+          "vnetName": { "value": "[parameters('vnetName')]" },
+          "subnets": { "value": "[reference(parameters('vnetName')).outputs.vnetSubnets.value]" },
+          "storageAccountName": { "value": "[reference('LabStorage').outputs.storageAccountName.value]" },
+          "availabilitySetName": { "value": "AVSetDC" },
+          "domainName": { "value": "[parameters('domainName')]" },
+          "adminUsername": { "value": "[parameters('adminUserName')]" },
+          "adminPassword": { "value": "[parameters('adminPassword')]" },
+          "domainJoinUsername": { "value": "[parameters('adminUserName')]" },
+          "domainJoinPassword": { "value": "[parameters('adminPassword')]" },
+          "assetLocation": { "value": "[parameters('assetLocation')]" },
+          "dDisks": {
+            "value": [
```

Comment:
Hi Kay,

Each member server has its own data disk. So, for 3 subnets, there are 2 member servers per subnet which means 6 data disks. This can be easily adapted: For example, if a user does not want a datadisk, he passes and empty array. (or do not pass the parameter, it has the default value in the linked template).

Brgds,

From: Kay Singh [mailto:notifications@github.com]
Sent: Thursday, February 18, 2016 12:36 AM
To: Azure/azure-quickstart-templates <azure-quickstart-templates@noreply.github.com>
Cc: Jean-Francois Six <jean-francois.six@microsoft.com>
Subject: Re: [azure-quickstart-templates] 301 subnet driven deployment (#1184)


In 301-subnet-driven-deployment/azuredeploy.json<https://github.com/Azure/azure-quickstart-templates/pull/1184#discussion_r53250391>:

> +          "lbName": { "value": "[parameters('lbName')]" },

> +          "computerName": { "value": "srvMbmr" },

> +          "publicStartRdpPort": { "value": 6000 },

> +          "publicIPAddressName": { "value": "[parameters('publicIPAddressName')]" },

> +          "vnetName": { "value": "[parameters('vnetName')]" },

> +          "subnets": { "value": "[reference(parameters('vnetName')).outputs.vnetSubnets.value]" },

> +          "storageAccountName": { "value": "[reference('LabStorage').outputs.storageAccountName.value]" },

> +          "availabilitySetName": { "value": "AVSetDC" },

> +          "domainName": { "value": "[parameters('domainName')]" },

> +          "adminUsername": { "value": "[parameters('adminUserName')]" },

> +          "adminPassword": { "value": "[parameters('adminPassword')]" },

> +          "domainJoinUsername": { "value": "[parameters('adminUserName')]" },

> +          "domainJoinPassword": { "value": "[parameters('adminPassword')]" },

> +          "assetLocation": { "value": "[parameters('assetLocation')]" },

> +          "dDisks": {

> +            "value": [

How does this work? With three subnets by default, does it create 6 data disks or a data disk with 6 in the name? I think it's the latter.

—
Reply to this email directly or view it on GitHub<https://github.com/Azure/azure-quickstart-templates/pull/1184/files#r53250391>.